src/components/List.js

```diff
@@ -26,10 +28,47 @@ const List = ({ token }) => {
   };
 
   const handleCheckbox = async (event) => {
-    const currentDate = Date.now(); // changed to using Date.now(), time given in milliseconds
+    const currentDate = Date.now(); // time given in milliseconds
     const queryCollection = await db.collection(token).doc(event.target.id);
+    // filteredListItem is our current snapshot of the doc
+    const filteredListItem = docs.filter(
+      (doc) => doc.id === queryCollection.id,
+    );
+    const listItem = filteredListItem[0];
+    const previouslyPurchasedDate = listItem.lastPurchased;
+    const latestInterval = currentDate - previouslyPurchasedDate;
+    // If item has not been purchased set to 1, else increment by 1
+    const numberOfPurchases = !listItem.numberOfPurchases
+      ? 1
+      : (listItem.numberOfPurchases += 1);
+    const estimate = calculateEstimate(
+      listItem.timeFrame,
+      latestInterval,
+      numberOfPurchases,
+    );
+
+    // FOR THE DEMO
+    // ------------------------------------------------------------------------------------
+    // update day of demo
+    // const oneWeekAgo = getTime(new Date(2021, 0, 26, 11, 30, 30));
+    // oneWeekAgo simulates the last date of purchase for the item
+    // console.log("Date a week ago: ", toDate(new Date(oneWeekAgo)));
+    // const demoLatestInterval = currentDate - oneWeekAgo;
+    // const exampleNumberOfPurchases = 7;
+    // const demoEstimate = calculateEstimate(
+    //   targetDoc[0].timeFrame,
+    //   demoLatestInterval,
+    //   exampleNumberOfPurchases,
+    // );
+    // demo estimate tells how long from purchase date until we're expected to buy it again
+    // console.log('Demo estimate :', addMilliseconds(previouslyPurchasedDate, demoEstimate));
+    // ------------------------------------------------------------------------------------
+    // END OF DEMO
+
     queryCollection.update({
       lastPurchased: currentDate,
+      estimatedNextPurchase: estimate,
+      numberOfPurchases: numberOfPurchases,
     });
   };
```

Comment:
This is a super-tricky one, and I'm trying to understand it, but right now, the estimatedNextPurchase is being saved as milliseconds, correct? Right now this is one of the values that I saved: 1612403611806, which converts to around 18,000 days or 01/13/53065. 

![image](https://user-images.githubusercontent.com/10378331/106837070-f3043e80-665f-11eb-8d8c-34d14ea3a46a.png)

I wonder if the three params for the estimates function above need to be 

1. `estimatedNextPurchase || timeFrame` in _days_ (use timeFrame if there hasn't been a calculated value yet
2. latest interval should be the distance between last purchase and this purchase in _days_
3. Number of purchases
Then it should return a number in _days_

I 100% could be wrong.

Reply:
Hmm...If I leave off the last three digits...then it's OK? You know what would be awesome...could you display the estimated value in the # of days next to the item name? Then it would be easier to understand and your reviewer would be super grateful : ). 👍

Reply:
That's a great idea, and maybe we could add a screenshot of some of the values we're checking with those comments, too, so you can see them before our demo.

Reply:
We added functionality to display the duration of time until the next purchase, but had some issues trying to display the actual date of the next purchase. After spending a reasonable amount of time trying to get it to get it to work we decided to move on. 
<img width="777" alt="timeTillNextPurchase-consoleLogs" src="https://user-images.githubusercontent.com/36529777/106970240-4aa6b680-6701-11eb-978f-22020b56a29f.PNG">